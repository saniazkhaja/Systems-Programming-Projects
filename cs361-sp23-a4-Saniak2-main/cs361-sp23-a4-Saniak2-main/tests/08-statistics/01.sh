#!/bin/bash

PORT=$@

REQUEST=$'GET /stats HTTP/1.1\r\n'
PING=$'GET /ping HTTP/1.1\r\n\r\n'

echo -n "$REQUEST" | nc 127.0.0.1 $PORT #0
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #1
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #2
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #3
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #4
echo ""
echo -n "$PING" | nc 127.0.0.1 $PORT &> /dev/null #5
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #6
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null #7
echo -n "$REQUEST" | nc 127.0.0.1 $PORT #8
echo ""
echo -n "$PING" | nc 127.0.0.1 $PORT &> /dev/null #5
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/read &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/05-files/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/Makefile &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/runtests.sh &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-files/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-error/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-error/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
