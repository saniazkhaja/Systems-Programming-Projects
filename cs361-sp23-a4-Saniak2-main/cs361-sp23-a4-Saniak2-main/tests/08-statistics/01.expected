HTTP/1.1 200 OK
Content-Length: 66

Requests: 0
Header bytes: 0
Body bytes: 0
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 68

Requests: 1
Header bytes: 39
Body bytes: 66
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 69

Requests: 2
Header bytes: 78
Body bytes: 134
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 3
Header bytes: 117
Body bytes: 203
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 4
Header bytes: 156
Body bytes: 273
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 6
Header bytes: 233
Body bytes: 347
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 8
Header bytes: 311
Body bytes: 476
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 10
Header bytes: 388
Body bytes: 550
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 12
Header bytes: 466
Body bytes: 680
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 14
Header bytes: 544
Body bytes: 810
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 16
Header bytes: 621
Body bytes: 888
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 18
Header bytes: 699
Body bytes: 1018
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 20
Header bytes: 778
Body bytes: 1559
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 22
Header bytes: 857
Body bytes: 1822
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 24
Header bytes: 937
Body bytes: 5130
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 73

Requests: 25
Header bytes: 976
Body bytes: 5202
Errors: 1
Error bytes: 22
HTTP/1.1 200 OK
Content-Length: 74

Requests: 26
Header bytes: 1015
Body bytes: 5275
Errors: 2
Error bytes: 44
HTTP/1.1 200 OK
Content-Length: 74

Requests: 27
Header bytes: 1054
Body bytes: 5349
Errors: 3
Error bytes: 66
